File: src/main/java/helloSpire/actions/DamageToGoldAction.java
package helloSpire.actions;


import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.AbstractGameAction.ActionType;
import com.megacrit.cardcrawl.actions.AbstractGameAction.AttackEffect;
import com.megacrit.cardcrawl.actions.common.HealAction;
import com.megacrit.cardcrawl.actions.utility.WaitAction;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect;

public class DamageToGoldAction extends AbstractGameAction {
    private DamageInfo info;

    public DamageToGoldAction(AbstractCreature target, DamageInfo info, AttackEffect effect) {
        this.info = info;
        this.setValues(target, info);
        this.actionType = ActionType.DAMAGE;
        this.attackEffect = effect;
    }

    public void update() {
        if (this.duration == 0.5F) {
            AbstractDungeon.effectList.add(new FlashAtkImgEffect(this.target.hb.cX, this.target.hb.cY, this.attackEffect));
        }

        this.tickDuration();
        if (this.isDone) {
            this.target.damage(this.info);
            //this.addToTop(new HealAction(this.source, this.source, this.target.lastDamageTaken));
            //FlatGoldAction(AbstractCreature target, DamageInfo info, int goldAmount)
            if (this.target.lastDamageTaken > 0) {
                this.addToTop(new FlatGoldAction(this.target, this.info, this.target.lastDamageTaken));
                this.addToTop(new WaitAction(0.1F));
            }
        }

        if (AbstractDungeon.getCurrRoom().monsters.areMonstersBasicallyDead()) {
            AbstractDungeon.actionManager.clearPostCombatActions();
        }
    }

}

File: src/main/java/robTheSpire/actions/DamageToGoldAction.kt
package robTheSpire.actions

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.actions.utility.WaitAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect

class DamageToGoldAction(target: AbstractCreature?, private val info: DamageInfo, effect: AttackEffect?) : AbstractGameAction() {
    override fun update() {
        if (duration == 0.5f) {
            AbstractDungeon.effectList.add(FlashAtkImgEffect(target.hb.cX, target.hb.cY, attackEffect))
        }
        tickDuration()
        if (isDone) {
            target.damage(info)
            //this.addToTop(new HealAction(this.source, this.source, this.target.lastDamageTaken));
            //FlatGoldAction(AbstractCreature target, DamageInfo info, int goldAmount)
            if (target.lastDamageTaken > 0) {
                addToTop(FlatGoldAction(target, info, target.lastDamageTaken))
                addToTop(WaitAction(0.1f))
            }
        }
        if (AbstractDungeon.getCurrRoom().monsters.areMonstersBasicallyDead()) {
            AbstractDungeon.actionManager.clearPostCombatActions()
        }
    }

    init {
        this.setValues(target, info)
        actionType = ActionType.DAMAGE
        attackEffect = effect
    }
}